src/riscv-integration.adoc

```diff
@@ -1032,17 +1034,37 @@ The fields are identical to <<mtval>> for CHERI exceptions.
 
 The <<stval2>> register is an SXLEN-bit read-write register.
 
-The CSR address is 0x14b.
+The CSR address is 0x24b.
 
 When a fault is taken into S-mode <<stval2>> is updated as for <<mtval2>> for all CHERI memory access traps.
 It is set to zero in all other cases.
 
 NOTE: This is not a standard RISC-V CSR, but <<mtval2>> is. 0x14b is the regular location for the CSR.
 
+<<stval2>> holds the same set of values that <<mtval2>> can hold.
+
 .Supervisor trap value register 2
 [#stval2-format]
 include::img/stval2reg.edn[]
 
+[#vstval2,reftext="vstval2"]
+==== Virtual Supervisor Trap Value Register 2 (vstval2)
+
+The <<vstval2>> register is an SXLEN-bit read-write register.
+
+The CSR address is 0x24b.
+
+When a fault is taken into VS-mode <<vstval2>> is updated as for <<mtval2>> for all CHERI memory access traps.
+It is set to zero in all other cases.
+
+NOTE: This is not a standard RISC-V CSR, but <<mtval2>> is. 0x24b is the regular location for the CSR.
+
+<<vstval2>> holds the same set of values that <<stval2>> can hold.
+
+.Virtual supervisor trap value register 2
+[#vstval2-format]
+include::img/vstval2reg.edn[]
```

Comment:
Maybe I missed it. Where is img/vstval2reg.edn ?

Reply:
I missed it - it's committed now - hopefully the build will work as a result